packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -55,17 +55,18 @@ const ProductDetail = () => {
     const childProductRefs = React.useRef({})
     const customerId = useCustomerId()
     /****************************** Basket *********************************/
-    const {data: basket} = useCurrentBasket()
+    const { data: basket } = useCurrentBasket()
     const addItemToBasketMutation = useShopperBasketsMutation('addItemToBasket')
-    const {res} = useServerContext()
+    const { res } = useServerContext()
     if (res) {
         res.set('Cache-Control', `max-age=${MAX_CACHE_AGE}`)
     }
+    let isBasketLoading = !!!basket?.basketId
```

Comment:
This is the part of the fix that determines if isBasketLoading is true or not. The check is for the existence of a basket id. If it's null, we don't have a basket yet.

Reply:
This has been updated and cleaned up to be

`const isBasketLoading = !basket?.basketId`